src/parser.rs

```diff
@@ -1741,12 +1783,77 @@ impl<'a> Parser<'a> {
                 if statement == until {
                     break
                 } else {
-                    statements.push(statement)
+                    match statement.clone() {
+                        Statement::Catch(code) => {
+                            match self.push_statement_or_return_fixed_code(
+                                statements,
+                                statement,
+                                code.clone(),
+                            ) {
+                                Some(fixed_code) => {
+                                    statements.push(Statement::Catch(fixed_code));
+                                }
+                                None => {}
+                            }
+                        }
+                        Statement::Else(code) => {
+                            match self.push_statement_or_return_fixed_code(
+                                statements,
+                                statement,
+                                code.clone(),
+                            ) {
+                                Some(fixed_code) => {
+                                    statements.push(Statement::Else(fixed_code));
+                                }
+                                None => {}
+                            }
+                        }
+                        Statement::ElseIf(condition, code) => {
+                            match self.push_statement_or_return_fixed_code(
```

Comment:
no need for this match (after the changes), you can just push the returned vec to the statements